### PR TITLE
EZP-19123: Make ezuser.login case in-sensitive across databases

### PR DESCRIPTION
**Pull request by andrerom**
https://jira.ez.no/browse/EZP-19123
Replaces #1527, needed for #1388 *(addition-of/switch-to bcrypt user password hashing)*.

Todo:
- [x] upgrade script

### CODE REVIEW

**Comment by bdunogier** [comment]
Looks clean to me. Is the other approach better in some aspects ?

**Comment by andrerom** [comment]
> Looks clean to me. Is the other approach better in some aspects ?

it was just easier, this one will need to deal with upgrade script.

**Comment by bdunogier** [comment]
If it is more correct then i'd rather go for it...

File: eZ/Publish/API/Repository/UserService.php [hunk 1]
```diff
@@ -148,8 +148,8 @@ public function loadAnonymousUser();
     /**
      * Loads a user for the given login and password.
      *
-     * This method is case sensitive in regards to $login and $password as a comparison of password hash is done
-     * which contains both parameters.
+     * Since 6.1 $login is handled case in-sensitive across all storage engines and database backends, however if login
```
Comment (joaoinacio) [inline eZ/Publish/API/Repository/UserService.php]: Nitpick: I believe it's `case insensitive` (no hyphen)

File: eZ/Publish/API/Repository/UserService.php [hunk 2]
```diff
@@ -164,7 +164,8 @@ public function loadUserByCredentials($login, $password);
     /**
      * Loads a user for the given login.
      *
-     * Note: This method loads user by $login where $login might be case in-sensitive on certain storage engines!
+     * Since 6.1 $login is handled case in-sensitive across all storage engines and database backends, like was the case
```
Comment (joaoinacio) [inline eZ/Publish/API/Repository/UserService.php]: same as above

**Comment by andrerom** [comment]
review ping @glye @joaoinacio 

*Note: Was previously attempting to normalize the characters, however did not find anything that does that for us easily and it's not within scope.*

File: eZ/Publish/API/Repository/UserService.php [hunk 3]
```diff
@@ -164,7 +164,8 @@ public function loadUserByCredentials($login, $password);
     /**
      * Loads a user for the given login.
      *
-     * Note: This method loads user by $login where $login might be case in-sensitive on certain storage engines!
+     * Since 6.1 $login is handled case insensitive across all storage engines and database backends, like was the case
```
Comment (bdunogier) [inline eZ/Publish/API/Repository/UserService.php]: Can be simplified to "login is case-insensitive".

**Comment by bdunogier** [comment]
Just a bit unsure about the "double wording": both "case insensitive" and "normalized" are used. Could we try to use only one of those ?

That aside, +1 on the implementation (but reviewed on mobile).

File: eZ/Publish/API/Repository/UserService.php (hunk 3 above)
Comment (andrerom) [inline eZ/Publish/API/Repository/UserService.php]: Ok, but regarding `case-insensitive` I changed this to `case insensitive` after feedback from Joao:
https://github.com/ezsystems/ezpublish-kernel/pull/1561#discussion_r50821173

**Comment by glye** [comment]
+1, esp. after the ADD COLUMN was added :)

File: data/update/postgres/unstable/dbupdate-6.1.0-to-6.2.0rc1.sql [hunk 4]
```diff
@@ -0,0 +1,9 @@
+-- Set storage engine schema version number
+UPDATE ezsite_data SET value='6.1.0' WHERE name='ezpublish-version';
```
Comment (andrerom) [inline data/update/postgres/unstable/dbupdate-6.1.0-to-6.2.0rc1.sql]: 6.2 :)

**Pull request merged**